### PR TITLE
Add scroll to top button

### PR DESCRIPTION
**Pull request by mojtabakaviani**
Show scroll to top button with animation, when the user scrolls more than 50% of the screen.

### CODE REVIEW

File: packages/components/src/components/DaScroll.vue [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    scrollTop() {
+      window.scrollTo({ top: 0, behavior: "smooth" });
+    },
+  },
+
+  created() {
+    this.scroll = (event) => {
+      this.showed =
+        document.documentElement.scrollTop >
+        document.documentElement.scrollHeight / 3;
```
Comment (idoshamun) [inline packages/components/src/components/DaScroll.vue:31]: I think it should be `/ 2` to make it 50%, right?
```suggestion
        document.documentElement.scrollHeight / 2;
```

File: packages/components/src/components/DaScroll.vue [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  </div>
+</template>
+
+<script>
+export default {
+  name: "DaScroll",
+
+  data() {
+    return {
+      showed: false,
```
Comment (idoshamun) [inline packages/components/src/components/DaScroll.vue:17]: A better naming would be `show`

File: packages/components/src/components/DaScroll.vue [hunk 3]
```diff
@@ -0,0 +1,80 @@
+<template>
+  <div
+    class="scroll__switch"
```
Comment (idoshamun) [inline packages/components/src/components/DaScroll.vue:3]: `scroll-btn` is a better naming for this, as this is not a switch component

File: packages/components/src/components/DaScroll.vue [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    import("../../icons/arrow");
+    window.addEventListener("scroll", this.scroll);
+  },
+};
+</script>
+
+<style>
+.scroll__switch {
+  height: 48px;
+  width: 48px;
```
Comment (idoshamun) [inline packages/components/src/components/DaScroll.vue:46]: width and height should be `64px` according to the specs

File: packages/components/src/components/DaScroll.vue [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  border-radius: 100%;
+  bottom: 32px;
+  right: 32px;
+  opacity: 0;
+  transition: 0.35s;
+  transform: scale(0.8);
```
Comment (idoshamun) [inline packages/components/src/components/DaScroll.vue:52]: No need to scale it down...

File: packages/components/src/components/DaScroll.vue [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  position: fixed;
+  background: var(--theme-primary);
+  visibility: hidden;
+
+  &:hover {
+    opacity: 1 !important;
```
Comment (idoshamun) [inline packages/components/src/components/DaScroll.vue:58]: Please don't use `!important`, it's not a good practice. It's better to increase specificity to make sure this rule is applied

File: packages/components/src/components/DaScroll.vue [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+      this.showed =
+        document.documentElement.scrollTop >
+        document.documentElement.scrollHeight / 3;
+    };
+    this.scroll = this.scroll.bind(this);
+  },
+
+  mounted() {
+    import("../../icons/arrow");
+    window.addEventListener("scroll", this.scroll);
```
Comment (idoshamun) [inline packages/components/src/components/DaScroll.vue:38]: Please make sure to use a passive event listener to improve scrolling performance.
Take a look here:
https://web.dev/uses-passive-event-listeners/

Also make sure to remove the event listener on dismount

**Review by idoshamun — CHANGES_REQUESTED** [review]
Thank you so much for this contribution 🙌
We do have some changes to work on and fix before we can merge it back to master.
Please also make sure to run `yarn lint --fix` in both components and extension folder to fix some of the lint issues that I saw.
In addition, please change the appear animation as follows:
https://codepen.io/ig_design/pen/yrwgwO

Let me know if you need further guidance 🙏

File: packages/components/src/components/DaScroll.vue (hunk 1 above)
Comment (mojtabakaviani) [reply packages/components/src/components/DaScroll.vue:31]: Your welcome, first how run service on windows or can use online service posts? second for 50% scroll when text devide by 3 is better!

File: packages/components/src/components/DaScroll.vue (hunk 4 above)
Comment (mojtabakaviani) [reply packages/components/src/components/DaScroll.vue:46]: Add progress animation?

File: packages/components/src/components/DaScroll.vue (hunk 1 above)
Comment (idoshamun) [reply packages/components/src/components/DaScroll.vue:27]: You can check out the instructions here:
https://github.com/dailydotdev/daily#-running-dailydev-locally

File: packages/components/src/components/DaScroll.vue (hunk 4 above)
Comment (idoshamun) [reply packages/components/src/components/DaScroll.vue:45]: No need for progress animation just a show and hide animation

**Comment by mojtabakaviani** [comment]
@idoshamun  please review changes again.

File: packages/components/src/components/DaScroll.vue [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+        > document.documentElement.scrollHeight / 3;
+    };
+    this.scroll = this.scroll.bind(this);
+  },
+
+  mounted() {
+    import('../../icons/arrow');
+    window.addEventListener('scroll', this.scroll, { passive: true });
+  },
+  dismount() {
```
Comment (idoshamun) [inline packages/components/src/components/DaScroll.vue:35]: It should be `beforeDestroy`

File: packages/components/src/components/DaScroll.vue [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    window.removeEventListener('scroll');
```
Comment (idoshamun) [inline packages/components/src/components/DaScroll.vue:36]: You forgot to provide the callback as a second argument

**Review by idoshamun — CHANGES_REQUESTED** [review]
See my comments.
Please also check it locally because I'm not sure it's perfectly fine.

File: packages/components/src/components/DaScroll.vue (hunk 1 above)
Comment (mojtabakaviani) [reply packages/components/src/components/DaScroll.vue:27]: when run daily-apps_daily-gateway_1 and daily-apps_daily-api_1 have no such file or directory error!

Comment (idoshamun) [reply packages/components/src/components/DaScroll.vue:27]: Let's discuss a separate issue. Please open a new issue and provide all the details.

**Comment by idoshamun** [comment]
@mojtabakaviani can you please give me permissions to push to this branch?
I would like to apply some changes before merging

**Comment by mojtabakaviani** [comment]
@idoshamun Sure! i invite you.

**Review by idoshamun — APPROVED** [review]

**Comment by idoshamun** [comment]
Thank you so much for this contribution! 🙏

**Comment by mojtabakaviani** [comment]
You're welcome! thank you to help me for this contribution. i will happy to continue contribution with you in the future!

**Pull request merged**